### PR TITLE
Anim component updates

### PR DESCRIPTION
**Pull request by ellt92**
- Fix for the `AnimComponent#stateGraphAsset` property
- Fix for the `AnimController#playable` property
- Dirty flag any materials that are animated
- Add the `AnimComponent#baseLayer` property as many workflows will only make use of a single layer. This stops users having to search for that layer using `AnimComponent#findAnimationLayer`.

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/framework/components/anim/component.js [hunk 1]
```diff
@@ -344,6 +345,21 @@ Object.assign(pc, function () {
...
             }
+        },
+        /**
+         * @private
+         * @name pc.AnimComponent#baseLayer
+         * @type {pc.AnimComponentLayer}
+         * @readonly
+         * @description Returns the base layer of the state graph
+         */
+        baseLayer: {
```
Comment (mvaligursky) [inline src/framework/components/anim/component.js:356]: good idea!

File: src/framework/components/anim/binder.js [hunk 2]
```diff
@@ -215,7 +215,13 @@ Object.assign(pc, function () {
                     propertyComponent[entityPropertySetterFunctionName](this._getProperty(propertyComponent, [entityProperty]));
                 };
                 return new pc.AnimTarget(entityPropertySetter.bind(this), animDataType, animDataComponents);
+            } else if (propertyHierarchy.indexOf('material') !== -1) {
+                return new pc.AnimTarget(function (values) {
+                    setter(values);
+                    propertyComponent.material.dirty = true;
```
Comment (mvaligursky) [inline src/framework/components/anim/binder.js:221]: I think it'd be safer to call material.update();

File: src/framework/components/anim/controller.js [hunk 3]
```diff
@@ -238,8 +238,8 @@ Object.assign(pc, function () {
             get: function () {
                 var playable = true;
                 var i;
-                for (i = 0; i < this._states.length; i++) {
-                    if (!this._states[i].playable) {
+                for (i = 0; i < this.states.length; i++) {
+                    if (!this._states[this.states[i]].playable) {
```
Comment (willeastcott) [inline src/framework/components/anim/controller.js:242]: This does look a bit odd. Are `this.states` really ids or names or something?

Comment (ellt92) [reply src/framework/components/anim/controller.js:242]: Yeah the controller exposes a list of available state names to the component but they're stored internally as AnimState objects. I could loop over `this._states` keys instead here if it makes things clearer.

Comment (ellt92) [reply src/framework/components/anim/controller.js]: Using `this._stateNames` here instead.

**Pull request merged**